internal/cmd/config.go

```diff
@@ -1986,7 +1986,9 @@ func (c *Config) persistentPreRunRootE(cmd *cobra.Command, args []string) error
 		c.sourceSystem = chezmoi.NewDryRunSystem(c.sourceSystem)
 		c.destSystem = chezmoi.NewDryRunSystem(c.destSystem)
 	}
-	if c.Verbose || annotations.hasTag(outputsDiff) {
+	if annotations.hasTag(outputsDiff) ||
+		c.Verbose &&
+			(annotations.hasTag(modifiesDestinationDirectory) || annotations.hasTag(modifiesSourceDirectory)) {
```

Comment:
You probably thought about this already, but let me suggest in case you didn't: have you thought about adding the `outputsDiff` tag to the command when applicable with Verbose? This way you could keep the if as `if annotations.hasTag(outputsDiff) {` only.

Reply:
The annotations are properties of the command, which are set before the flags are parsed.